src/vector/sources/eventstoredb/cluster.rs

```diff
@@ -0,0 +1,163 @@
+    let mut ticks = IntervalStream::new(tokio::time::interval(Duration::from_secs(
+        config.scrape_interval_secs,
+    )))
+    .take_until(cx.shutdown);
+
+    let namespace = config
+        .default_namespace
+        .clone()
+        .unwrap_or_else(|| "eventstoredb".to_string());
+
+    Ok(Box::pin(
+        async move {
+            let mut epoch_number = i64::MAX;
+            let mut gen = rand::rngs::SmallRng::from_entropy();
+
+            while ticks.next().await.is_some() {
+                match client.read_gossip().await {
+                    Err(error) => {
+                        tracing::error!(target: "eventstoredb_nexus_cluster_metrics", "{}", error)
+                    }
+                    Ok(members) => {
+                        let now = chrono::Utc::now();
+                        let tags = BTreeMap::new();
+                        let mut metrics = Vec::new();
+                        let dead_count = members.iter().filter(|m| !m.is_alive).count();
+                        let leaders = members
+                            .iter()
+                            .filter(|m| m.state == VNodeState::Leader)
+                            .collect::<Vec<&MemberInfo>>();
+
+                        if leaders.is_empty() {
+                            metrics.push(
+                                Metric::new(
+                                    "leader_mismatches",
+                                    MetricKind::Absolute,
+                                    MetricValue::Counter { value: 1.0 },
+                                )
+                                .with_namespace(Some(namespace.clone()))
+                                .with_tags(Some(tags.clone()))
+                                .with_timestamp(Some(now)),
+                            );
+                        }
+
+                        if dead_count > 0 {
+                            metrics.push(
+                                Metric::new(
+                                    "dead_nodes",
+                                    MetricKind::Absolute,
+                                    MetricValue::Counter {
+                                        value: dead_count as f64,
+                                    },
+                                )
+                                .with_namespace(Some(namespace.clone()))
+                                .with_tags(Some(tags.clone()))
+                                .with_timestamp(Some(now)),
+                            );
+                        }
+
+                        let current_epoch_number = if leaders.len() == 1 {
+                            leaders[0].epoch_number
+                        } else {
+                            let idx = gen.next_u32() % members.len() as u32;
+                            members[idx as usize].epoch_number
+                        };
+
+                        if epoch_number == i64::MAX {
```

Comment:
Why not just make this an `Option<i64>` that starts out set to none?

Reply:
Because it would be a pain to use with no extra benefit.

Reply:
Couldn't it just be:

```rust
match epoch_number {
        None => {
            epoch_number = Some(current_epoch_number);
        },
        Some(value) => {
            if value != current_epoch_number {
                epoch_number = Some(current_epoch);
                metrics.push( // push the metric, etc etc
            }        
        }
    }
    
    if metrics.is_empty() {  
        continue;
    } // rest as before
```

Reply:
It's an extra line or two, but it would be more readable. I've also read and written code for instance that treats `-1` as an invalid number / error condition, but anytime I had an option type available (like boost::optional in C++ or `typing.Option[int]` in Python) I've used that instead because it's makes the idiom more obvious.

Reply:
Done